modules/seqexec/server/src/main/scala/seqexec/server/GiapiInstrumentController.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server
+
+import cats.data.EitherT
+import cats.effect.Sync
+import cats.implicits._
+import giapi.client.GiapiClient
+import giapi.client.commands.{CommandResult, CommandResultException, Configuration}
+import org.log4s.getLogger
+import seqexec.model.dhs.ImageFileId
+import seqexec.server.SeqexecFailure.{Execution, SeqexecException}
+import seqexec.server.keywords.GdsClient
+import squants.time.Time
+
+import scala.concurrent.duration._
+
+/**
+  * Superclass for all GIAPI instrument controllers.
+  */
+abstract class GiapiInstrumentController[F[_]: Sync, CFG, C <: GiapiClient[F]] {
```

Comment:
Do you really need to restrict to `Sync` here?

Reply:
My understanding is yes, although I'm not sure why. If I remove the `Sync`, it curries, adding something like an `F => Sync` at the end. (I'm not 100% sure that that's it. I'll check and reply.)